GG/src/ListBox.cpp

```diff
@@ -1533,25 +1538,29 @@ void ListBox::KeyPress(Key key, boost::uint32_t key_code_point, Flags<ModKey> mo
 
         // horizontal scrolling keys
         case GGK_LEFT: // left key (not numpad key)
-            if (m_first_col_shown) {
-                --m_first_col_shown;
-                m_hscroll->ScrollTo(
-                    Value(std::accumulate(m_col_widths.begin(), m_col_widths.begin() + m_first_col_shown, X0)));
-                SignalScroll(*m_hscroll, true);
-            }
+            if (m_first_col_shown == 0)
+                break;
+
+            --m_first_col_shown;
```

Comment:
could do with some comments to explain this rather large mess of nested calls... or can you use some temporaries to make it more readable?